table/metadata.go

```diff
@@ -610,7 +621,12 @@ func (b *MetadataBuilder) SetLastUpdatedMS() *MetadataBuilder {
 	return b
 }
 
-func (b *MetadataBuilder) buildCommonMetadata() *commonMetadata {
+func (b *MetadataBuilder) buildCommonMetadata() (*commonMetadata, error) {
+	if b.defaultSpecID == nil {
+		return nil, errors.New("no default spec ID specified")
+	}
```

Comment:
doesn't the iceberg spec say that the default specID is just 0?

Reply:
Thanks, good catch, just checked the java code and there's no requirement to set the default. It's however validated to exist & be compatible with the current schema in `build()`:

`PartitionSpec.checkCompatibility(specsById.get(defaultSpecId), schema);`